data_management/autopreprocess_testing/preprocess_directory.py

```diff
@@ -1,34 +1,45 @@
 """
 Base code for batch pre-processing
 1. Gets file from directory
-2. Does some pre-processing on it
+2. Does pre-processing on the file
 3. Puts processed file in new directory
 """
-
+import sys
+import queue
 import cv2
 import os
 import glob
 from PIL import Image
-from processing_pipeline import processing_pipeline
+from processing_pipeline import processing_pipeline 
+
+# Get list of all image paths
+def get_all_images(source_dir):
+
+    supported_file_extensions = ["jpg", "jpeg", "png", "tif", "tiff"]
+    img_files = []
+    for ext in supported_file_extensions:
+        img_files += glob.glob(source_dir + f"/**/*.{ext}", recursive=True)
+    return img_files
 
 # Change to current directory
 os.chdir(os.path.dirname(os.path.abspath(__file__)))
-# Test image directory
-source_dir = os.getcwd() + '/test_images'
-# Processed image directory
-target_dir = os.getcwd() + '/processed_test_images/'
+# Test image directory;
 
-# Get list of all image paths
-supported_file_extensions = ["jpg", "jpeg", "png", "tif", "tiff"]
-img_files = []
-for ext in supported_file_extensions:
-    img_files += glob.glob(source_dir + f"/**/*.{ext}", recursive=True)
+default_directory = '/test_images22622/'
+if len(sys.argv) > 1:
+    default_directory = sys.argv[1]
+
+source_dir = os.getcwd() + default_directory
+# Processed image directory, I also changed the file folder, user needs to change the folder to the one they wish to train images on.
+target_dir = os.getcwd() + '/processed_test_images/'
```

Comment:
same as other path joining

Reply:
Changed concatenated paths to os.path.join(path1, path2...) syntax.  This enables one to be platform agnostic when running the preprocessing function.